server/src/main/java/org/opensearch/gateway/remote/ClusterStateDiffManifest.java

```diff
@@ -0,0 +1,604 @@
+import org.opensearch.core.xcontent.MediaTypeRegistry;
+import org.opensearch.core.xcontent.ToXContentObject;
+import org.opensearch.core.xcontent.XContentBuilder;
+import org.opensearch.core.xcontent.XContentParseException;
+import org.opensearch.core.xcontent.XContentParser;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import static org.opensearch.core.xcontent.XContentParserUtils.ensureExpectedToken;
+
+/**
+ * Manifest of diff between two cluster states
+ *
+ * @opensearch.internal
+ */
+public class ClusterStateDiffManifest implements ToXContentObject {
+    private static final String FROM_STATE_UUID_FIELD = "from_state_uuid";
+    private static final String TO_STATE_UUID_FIELD = "to_state_uuid";
+    private static final String METADATA_DIFF_FIELD = "metadata_diff";
+    private static final String COORDINATION_METADATA_UPDATED_FIELD = "coordination_metadata_diff";
+    private static final String SETTINGS_METADATA_UPDATED_FIELD = "settings_metadata_diff";
+    private static final String TRANSIENT_SETTINGS_METADATA_UPDATED_FIELD = "transient_settings_metadata_diff";
+    private static final String TEMPLATES_METADATA_UPDATED_FIELD = "templates_metadata_diff";
+    private static final String HASHES_OF_CONSISTENT_SETTINGS_UPDATED_FIELD = "hashes_of_consistent_settings_diff";
+    private static final String INDICES_DIFF_FIELD = "indices_diff";
+    private static final String METADATA_CUSTOM_DIFF_FIELD = "metadata_custom_diff";
+    private static final String UPSERTS_FIELD = "upserts";
+    private static final String DELETES_FIELD = "deletes";
+    private static final String CLUSTER_BLOCKS_UPDATED_FIELD = "cluster_blocks_diff";
+    private static final String DISCOVERY_NODES_UPDATED_FIELD = "discovery_nodes_diff";
+    private static final String ROUTING_TABLE_DIFF = "routing_table_diff";
+    private static final String CLUSTER_STATE_CUSTOM_DIFF_FIELD = "cluster_state_custom_diff";
+
+    private final String fromStateUUID;
+    private final String toStateUUID;
+    private final boolean coordinationMetadataUpdated;
+    private final boolean settingsMetadataUpdated;
+    private final boolean transientSettingsMetadataUpdate;
+    private final boolean templatesMetadataUpdated;
+    private List<String> customMetadataUpdated;
+    private final List<String> customMetadataDeleted;
+    private final List<String> indicesUpdated;
+    private final List<String> indicesDeleted;
+    private final boolean clusterBlocksUpdated;
+    private final boolean discoveryNodesUpdated;
+    private final List<String> indicesRoutingUpdated;
+    private final List<String> indicesRoutingDeleted;
+    private final boolean hashesOfConsistentSettingsUpdated;
+    private final List<String> clusterStateCustomUpdated;
+    private final List<String> clusterStateCustomDeleted;
+
+    ClusterStateDiffManifest(ClusterState state, ClusterState previousState) {
```

Comment:
What is the difference between this and ClusterMetadataManifest?

Reply:
This classes contains the diff information for cluster state. This is added as a member under ClusterMetadataManifest and will be used to determine what objects are to be downloaded to apply the diff cluster state.